migration/src/main/java/com/intershop/customization/migration/pfconfigurationfs/ConfigurationXMLBuilder.java

```diff
@@ -0,0 +1,246 @@
+package com.intershop.customization.migration.pfconfigurationfs;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import com.intershop.customization.migration.pfconfigurationfs.CfgResourceConverter.ResourceType;
+
+/**
+ * This method would contain logic to build the XML configuration
+ * based on the cartridge name and the provided configuration XML.
+* The implementation details are not provided in the original code snippet.
+*/
+
+public class ConfigurationXMLBuilder {
+
+    private String cartridgeName = "";
+
+    private ArrayList<String> headerLines = new ArrayList<>();
+    private ArrayList<String> footerLines = new ArrayList<>();
+    private ArrayList<String> lines = new ArrayList<>();
+    private ArrayList<String> commonLines = new ArrayList<>();
+
+    // keep the file names already added to the XML
+    private HashSet <String> domainResources = new HashSet<>();
+    private HashSet <String> commondResources = new HashSet<>();
```

Comment:
my comment is somehow gone, but was not resolved

```suggestion
    private List<String> headerLines = new ArrayList<>();
    private List<String> footerLines = new ArrayList<>();
    private List<String> lines = new ArrayList<>();
    private List<String> commonLines = new ArrayList<>();

    // keep the file names already added to the XML
    private Set <String> domainResources = new HashSet<>();
    private Set <String> commondResources = new HashSet<>();
```